### DISCUSSION TITLE
populating not work with many to one with select in but work with join strategy

### DISCUSSION THREAD

**Discussion by n4an** [mikro-orm/mikro-orm#2667] [Q&A]
**Describe the bug**

**Stack trace**
```
...
```

**To Reproduce**

With lazy equal to true not work if join strategy is select_in

  @ManyToOne(() => User, { cascade: [Cascade.ALL], lazy: true, wrappedReference: true })
  user: IdentifiedReference<User, 'id'>

return await this.usersService.findAllUsers(
        {
          isActive: true
        },
        {
          populate: [
            'profiles',
            'roles.permissions'
          ],
          // strategy: LoadStrategy.JOINED
        }
)

With join works.
I changet that little https://github.com/n4an/nest-mikro-rom (join strategy and lazy)

**Expected behavior**

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 16 |
| typescript | 3.8.3 |
| mikro-orm | 5.0.0-dev.664 |
| your-driver | mysql |

**Comment by B4nan** [comment]
If you use lazy property, you are the one who needs to mark it as loaded - which you are not doing. With joined strategy you dont really need to select that column, as there is just one query - we need it for population only with select-in strategy.

Lazy flag is mainly for scalar properties, relations are lazy by definition, so I would say just drop the `lazy: true`. It controls what gets selected (so whether we add the FK to select clause), it has nothing to do with populating relations.

Also, what does "it works" and "not work" even mean? Be specific.
